atlas-core/src/main/java/io/atlasmap/actions/NumberFieldActions.java

```diff
@@ -168,6 +169,81 @@
         volumeConvertionTable = Collections.unmodifiableMap(rootTable);
     }
 
+    @AtlasFieldActionInfo(name = "AbsoluteValue", sourceType = FieldType.NUMBER, targetType = FieldType.NUMBER, sourceCollectionType = CollectionType.NONE, targetCollectionType = CollectionType.NONE)
+    public static Number absoluteValue(Action action, Number input) {
+        if (input == null) {
+            return 0;
+        }
+        if (input instanceof BigDecimal) {
+            return ((BigDecimal) input).abs();
+        } else if (input instanceof Double) {
+            return Math.abs(input.doubleValue());
+        } else if (input instanceof Float) {
+            return Math.abs(input.floatValue());
+        } else if (input instanceof Long) {
+            return Math.abs(input.longValue());
+        } else if (input instanceof AtomicLong) {
+            return Math.abs(((AtomicLong) input).get());
+        } else if (input instanceof Integer) {
+            return Math.abs(input.intValue());
+        } else if (input instanceof AtomicInteger) {
+            return Math.abs(((AtomicInteger) input).get());
+        } else if (input instanceof Byte) {
+            return (byte) Math.abs(input.byteValue());
+        } else {
+            return Math.abs(Double.parseDouble(input.toString()));
+        }
+    }
+
+    @AtlasFieldActionInfo(name = "Average", sourceType = FieldType.COMPLEX, targetType = FieldType.NUMBER, sourceCollectionType = CollectionType.ALL, targetCollectionType = CollectionType.NONE)
```

Comment:
I think `COMPLEX`  as a field type doesn't work, as UI doesn't allow mapping COMPLEX field (it just expands instead). Generally number field action should be NUMBER > NUMBER action, and prerequisite/subsequent type converter should be handled by the `TypeConverter`.